### ISSUE TITLE
[rstmgr] module reset detection improvements

### ISSUE THREAD

**Issue by tjaychen** [lowRISC/opentitan#7780]
We should improve the reset detection scheme to be automatic.
Right now, the resets for each module is manually declared, and the user needs to make sure to pick both the reset with the right root and the right clock domain.

The second part can be error prone and has seen issues surfaced in a few places.

However, given that clocks / resets are always defined as a pair in module hjson, we should just ask the top level declaration to declare the root reset, and auto detect what the appropriate clock domain would be. 

This would make the construction correct by design instead of requiring detailed user input.

**Comment by tjaychen** [comment]
@msfschaffner 
@eunchan

**Comment by moidx** [comment]
CC @tjaychen 

Moving to backlog as this is P3, Future Release

**Comment by msfschaffner** [comment]
CC @matutem

**Comment by matutem** [comment]
I don't think it is sensible to make this for PROD, since the hjson files are stable and working. Perhaps we can look into this for multi-top, but I would not give it a high-priority since there is so much more work that is essential for multi-top.

**Comment by msfschaffner** [comment]
Agreed, this is out of scope for PROD.
